src/category_theory/monad/equiv_mon.lean

```diff
@@ -0,0 +1,233 @@
+def to_Mon : Monad C → Mon_ (C ⥤ C) := λ M,
+{ X := M.func,
+  one := η_ _,
+  mul := μ_ _,
+  one_mul' := begin
+    change (_ ◫ _) ≫ _ = _,
+    ext A,
+    simp only [nat_trans.hcomp_id_app, nat_trans.comp_app],
+    apply monad.right_unit,
+  end,
+  mul_one' := begin
+    change (_ ◫ _) ≫ _ = _,
+    tidy,
+  end,
+  mul_assoc' := begin
+    change (_ ◫ _) ≫ _ = _ ≫ (_ ◫ _) ≫ _,
+    ext A,
+    simp only [nat_trans.hcomp_id_app, nat_trans.hcomp_app, functor.map_id,
+      nat_trans.id_app, comp_id, nat_trans.comp_app],
+    erw id_comp,
+    simp_rw monad.assoc,
+    change _ = ((α_ M.func M.func M.func).app A).hom ≫ _ ≫ _,
+    suffices : ((α_ M.func M.func M.func).app A).hom = 𝟙 _, by {rw this, simp},
+    refl,
+  end }
+
+variable (C)
+/-- Passing from `Monad C` to `Mon_ (C ⥤ C)` is functorial. -/
+def Monad_to_Mon : Monad C ⥤ Mon_ (C ⥤ C) :=
+{ obj := to_Mon,
+  map := λ M N f,
+  { hom := f.to_nat_trans,
+    one_hom' := begin
+      ext,
+      simp only [nat_trans.comp_app],
+      apply f.app_η,
+    end,
+    mul_hom' := begin
+      change _ = (_ ◫ _) ≫ _,
+      ext,
+      simp only [nat_trans.hcomp_app, assoc, nat_trans.comp_app],
+      change (μ_ _).app x ≫ f.app x = _,
+      rw f.app_μ,
+      simp only [nat_trans.naturality, assoc],
+      refl,
+    end } }
+variable {C}
+
+/-- To every monoid object in `C ⥤ C` we associate a `Monad C`. -/
+def of_Mon : Mon_ (C ⥤ C) → Monad C := λ M,
+{ func := M.X,
+  str :=
+  { η := M.one,
+    μ := M.mul,
+    assoc' := begin
+      intro X,
+      rw ←nat_trans.hcomp_id_app,
+      change ((M.mul ◫ 𝟙 M.X) ≫ M.mul).app X = _,
+      erw M.mul_assoc,
+      simp only [nat_trans.comp_app],
+      change ((α_ M.X M.X M.X).app X).hom ≫ (_ ◫ _).app X ≫ _ = _,
+      suffices : ((α_ M.X M.X M.X).app X).hom = 𝟙 _, by {rw this, simp},
+      refl,
+    end,
+    left_unit' := begin
+      intro X,
+      have := M.mul_one,
+      change (_ ◫ _) ≫ _ = _ at this,
+      rw [←nat_trans.id_hcomp_app, ←nat_trans.comp_app, this],
+      refl,
+    end,
+    right_unit' := begin
+      intro X,
+      have := M.one_mul,
+      change (_ ◫ _) ≫ _ = _ at this,
+      rw [←nat_trans.hcomp_id_app, ←nat_trans.comp_app, this],
+      refl,
+    end } }
+
+variable (C)
+/-- Passing from `Mon_ (C ⥤ C)` to `Monad C` is functorial. -/
+def Mon_to_Monad : Mon_ (C ⥤ C) ⥤ Monad C :=
+{ obj := of_Mon,
+  map := λ M N f,
+  { app_η' := begin
+      intro X,
+      simp only [auto_param_eq],
+      erw [←nat_trans.comp_app,f.one_hom],
+      refl,
+    end,
+    app_μ' := begin
+      intro X,
+      simp only [auto_param_eq],
+      erw [←nat_trans.comp_app, f.mul_hom],
+      simp only [nat_trans.naturality, assoc, nat_trans.comp_app],
+      erw [nat_trans.hcomp_app, assoc],
+      refl,
+    end,
+    ..f.hom } }
+variable {C}
+
+theorem of_to_mon_end_obj (M : Mon_ (C ⥤ C)) : (Mon_to_Monad C ⋙ Monad_to_Mon C).obj M = M :=
+  by {apply Mon_.hext, repeat {refl}}
```

Comment:
```suggestion
by {apply Mon_.hext, repeat {refl}}
```
No need to indent proofs/definitions. (Just indent multiline statements, and inside `begin ... end` blocks.)